[nuclide-swift] Add typehints

modocache opened the pull request.

modocache committed:
[nuclide-swift] Add typehints

modocache commented on pkg/nuclide-swift/lib/buildsystem/typehint/SwiftPMTypeHintProvider.js:
```diff
@@ -0,0 +1,78 @@
+  }
+
+  async typeHint(editor: TextEditor, position: atom$Point): Promise<?TypeHint> {
+    console.log('FIXME: Why is this console.log never printed...?');
```
@nmote I've verified that `createTypeHintProvider()` below is called. From what I understand about typehints, I should see this console statement printed whenever I hover over an identifier in a Swift source file, but this method never seems to get called. Can you spot what I'm doing wrong here? Any advice would be appreciated -- thanks!

nmote commented on pkg/nuclide-swift/lib/main.js:
```diff
@@ -98,6 +99,17 @@ export function createAutocompleteProvider(): atom$AutocompleteProvider {
 
+export function createTypeHintProvider(): TypeHintProvider {
+  return {
+    selector: '.source.swift',
```
The leading dot here needs to be taken away. I think autocomplete is the only one that requires a leading dot, as frustrating as that is.

modocache committed:
[nuclide-swift] Use `sourcekitten structure` for typehints

This works, and is pretty good. But it'll be even better once we start using
`sourcekitten structure` instead.

modocache committed:
[nuclide-swift] Use `sourcekitten index` for typehints
